logbook-core/src/main/java/org/zalando/logbook/JsonHttpLogFormatter.java

```diff
@@ -133,19 +125,20 @@ private Object tryParseBodyAsJson(final String body) {
         return body;
     }
 
-    private boolean isJson(final String contentType) {
-        if (contentType.isEmpty()) {
+    private boolean isJson(final String type) {
+        if (type.isEmpty()) {
             return false;
         }
 
-        final MediaType mediaType = MediaType.parse(contentType);
+        final int pi = type.indexOf(';');
+        final int limit = (pi != -1 ? pi : type.length());
 
-        final boolean isJson = mediaType.is(APPLICATION_JSON);
-
-        final boolean isApplication = mediaType.is(MediaType.ANY_APPLICATION_TYPE);
-        final boolean isCustomJson = mediaType.subtype().endsWith("+json");
+        return type.regionMatches(true, 0, "application/json", 0, limit)
```

Comment:
> According to regex101.com it would incorrectly match application/xml; incompatible=+json

Good catch. The definition of tokens and delimiters is here https://tools.ietf.org/html/rfc7230#section-3.2.6 but I think matching everything except space and semicolon should work.

    private static final Pattern MEDIA_TYPE_JSON_PATTERN = Pattern.compile("application/(?:json|[^\t ;]+\\+json)(?:$|\t| |;)");